avm/res/network/express-route-gateway/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,78 @@
+targetScope = 'subscription'
+
+metadata name = 'WAF-aligned'
+metadata description = 'This instance deploys the module in alignment with the best-practices of the Azure Well-Architected Framework.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-network.expressRouteGateway-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'nergwaf'
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableTelemetry bool = true
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-nestedDependencies'
+  params: {
+    virtualWANName: 'dep-${namePrefix}-vwan-${serviceShort}'
+    virtualHubName: 'dep-${namePrefix}-hub-${serviceShort}'
+    location: location
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+  }
+}
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../../main.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-test-${serviceShort}'
+  params: {
+    enableTelemetry: enableTelemetry
+    name: '${namePrefix}${serviceShort}001'
+    location: location
+    tags: {
+      'hidden-title': 'This is visible in the resource name'
+      hello: 'world'
+    }
+    autoScaleConfigurationBoundsMin: 2
+    autoScaleConfigurationBoundsMax: 3
+    virtualHubId: nestedDependencies.outputs.virtualHubResourceId
+    lock: {
+      kind: 'CanNotDelete'
+      name: 'myCustomLockName'
+    }
+    roleAssignments: [
```

Comment:
Can be removed and with it the MSI deployment in the dependencies.

Note: Required a refresh of the readme